dask/order.py

```diff
@@ -736,10 +684,7 @@ def finish_now_key(x):
 
                         inner_stack_pop = inner_stack.pop
                     for key, vals in dep_pools.items():
-                        if key < item_key:
-                            next_nodes[key].append(vals)
-                        else:
-                            later_nodes[key].append(vals)
+                        next_nodes[key].append(vals)
```

Comment:
- [ ] TODO: Considering replicating possible_singles logic here but don't have tests for it